File: fizz-buzz/fizzbuzz.lua
local function is_divided(num, divisor)
  return num % divisor == 0
end

local function fizzbuzz(num)
  if is_divided(num, 15) then
    return "FizzBuzz"
  end

  if is_divided(num, 3) then
    return "Fizz"
  end

  if is_divided(num, 5) then
    return "Buzz"
  end

  return tostring(num)
end

return fizzbuzz

File: fizz-buzz/fizzbuzz.go
package main

import "strconv"

func isDivisible(value, divisor int) bool {
	return value%divisor == 0
}

func FizzBuzz(value int) string {
	if isDivisible(value, 15) {
		return "FizzBuzz"
	}
	if isDivisible(value, 5) {
		return "Buzz"
	}
	if isDivisible(value, 3) {
		return "Fizz"
	}
	return strconv.Itoa(value)
}
